src/wowman/core.clj

```diff
@@ -326,8 +342,15 @@
   [file-opts map?, cli-opts map?]
   (debug "loading file config:" file-opts)
   (let [default-cfg (:cfg -state-template)
+
+        ;; todo: this sucks,fix
+        file-opts (if (contains? file-opts :selected-catalog)
+                    (update-in file-opts [:selected-catalog] keyword)
+                    file-opts)
+
         cfg (merge default-cfg file-opts)
         cfg (handle-legacy-install-dir cfg)
+        ;; doesn't support optional :opt keysets
```

Comment:
can we fetch the spec, inspect the required/optional keys and remove the unsupported ones ourselves? we could then drop this spec-tools as a dependency (I think)

Reply:
added to todo